fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckNetworkMiddleware.kt

```diff
@@ -83,6 +83,13 @@ class ReviewQualityCheckNetworkMiddleware(
                     store.onReanalyze()
                 }
 
+                ReviewQualityCheckAction.ReportProductBackInStock -> {
+                    val status = reviewQualityCheckService.reportBackInStock()
+                    if (status == ReportBackInStockStatusDto.REPORT_CREATED) {
+                        store.updateProductReviewState(ProductReviewState.Error.ThanksForReporting)
+                    }
```

Comment:
This is fine for now but we should consider a future improvement here. 

When the user clicks the button, nothing really happens on the ui till the network request succeeds with the right response and then the UI transitions to thanks for reporting. Maybe we should add a loading state here while the request is inflight.

Reply:
Or we should treat this as a success by default and update to `ThanksForReporting` immediately, and only revert back to the previous state in case of error.

Reply:
It seems like that's already happening in the Store reducer, which now makes me wonder why do we need `store.updateProductReviewState(ProductReviewState.Error.ThanksForReporting)` ?

Reply:
hm ya we do update to `ThanksForReporting` immediately, I removed it here

Reply:
I'm not sure we should not update the state to `NoAnalysisPresent()` directly here. By just doing this, we are expecting that the report product api responses are immediately in sync with the analysis api.  This case should hardly ever happen. 

I think for the `productReviewState` we should use the analysis api as the source of truth  i.e fire the Fetch action that will take care of fetching and mapping, so the mapper does the usual checks and maps to the correct state - which could be `NoAnalysisPresent`. wdyt?

Reply:
Hm okay, but won't fetching and calling the mapper still map to one of the errors Error.ProductAlreadyReported or ProductReviewState.Error.ProductNotAvailable?

Reply:
```
if (status == ReportBackInStockStatusDto.NOT_DELETED) {
    // fetch and map
}
```

Since this will happen when the report api returns not deleted, I doubt that the analysis api will return a response that leads to Error.ProductAlreadyReported or ProductReviewState.Error.ProductNotAvailable. 

If that does indeed happen, it would also happen if we close and open the sheet as that would trigger the fetch and map again.